### ISSUE TITLE
Separate IID and grouped splits in RamanBench results

### ISSUE THREAD

**Issue by mario-koddenbrock** [ml-lab-htw/RamanBench#6]
`RamanBenchmark._grouped_train_test_split` (`src/raman_bench/benchmark.py`) currently makes an **implicit, invisible choice** per key:

```python
groups = np.array(group_labels)
if len(np.unique(groups)) == len(data_df):
    return train_test_split(...)          # IID fallback
gss = GroupShuffleSplit(...)              # grouped
```

Two rows are grouped when their non-zero target values are identical (replicate measurements of the same physical sample). If every row is its own group, the split silently degrades to a plain IID `train_test_split`.

This means the benchmark currently mixes **two fundamentally different evaluation regimes in one leaderboard**, with no way to tell them apart:

- **Grouped keys** — replicates are kept together; no leakage; harder, more realistic.
- **IID keys** — no grouping applies; standard random split.

Measured on the current benchmark (`configs/v1_hpo.json`, 150 comparable keys): roughly **half** take each path. Nothing in the results records which.

Two concrete problems this caused:

1. When diagnosing a split bug, the only way to identify which keys were grouped was to re-derive the split and compare — the results themselves carried no signal.
2. A model's Elo currently averages over both regimes. A model that does well on IID keys and poorly on grouped ones (i.e. relies on replicate leakage) is **indistinguishable** from a genuinely robust one.

## Goal

Record the split regime per key, and let downstream code filter on it.

### 1. Core: expose the regime (`RamanBench`)

- Have `_grouped_train_test_split` return (or record) which branch it took and the group statistics: `split_type` ∈ `{"iid", "grouped"}`, `n_groups`, `largest_group_size`.
- Surface it on the benchmark so `compute_predictions` can persist it. `n_groups == n_rows` ⟺ `iid` is the natural definition — it's exactly the existing fallback condition, so no new logic is needed, only reporting.
- Note: for classification, `_load_dataset_from_key` uses a stratified split and never enters the grouped path — decide whether that is a third value (`"stratified"`) or simply `"iid"`. Recommend a distinct value; conflating them loses information.

### 2. Persist it alongside results

Two options; prefer (a):

**(a) Per-key metadata file** — e.g. `results/<run>/seed_N/predictions/<key>_split_info.json` written next to the ground truth, containing `split_type`, `n_groups`, `n_train`, `n_test`. Cheap, additive, no schema break.

**(b) A column in the metrics CSVs** — `split_type` added in `compute_metrics_from_predictions`. More convenient downstream but changes the CSV schema.

⚠️ **Write it where the ground truth is written, not gated behind the already-complete check.** `compute_predictions` returns at the skip (`predictions.py:514`) *before* writing ground truth (line ~604), so metadata placed after that point would never appear for already-computed keys. Consider writing it during the metrics step instead, where the full key set is always traversed, or explicitly backfilling.

### 3. Downstream: filtering (`raman_bench_paper`)

- Add `--split-type {all,iid,grouped}` to `scripts/plot_rankings.py` and the Elo/leaderboard path, defaulting to `all` so existing behaviour is unchanged.
- Same filter in `scripts/generate_results_tables.py`.
- Ideally: report the leaderboard **split by regime** in the paper, or at minimum report `n_iid` / `n_grouped` so readers know the mix.

### 4. Backfill

The regime is a pure function of the data (given a fixed dataset version), so it can be derived for existing runs without recomputing predictions — iterate `configure_benchmark` and record which branch each key takes.

## Acceptance criteria

- [ ] Every key in a completed run has a recorded `split_type`.
- [ ] `split_type` is derivable **without** re-deriving the split.
- [ ] Leaderboard/Elo can be produced for `iid` only, `grouped` only, or both.
- [ ] Counts reported: how many keys of each regime.
- [ ] Backfill works on an existing run with no re-training.
- [ ] Classification keys are handled explicitly (not silently bucketed as `iid`).

## Out of scope

- Changing which keys are grouped, or the grouping rule itself.
- Re-running the benchmark.

## Notes / risks

- **Dataset-version dependence:** the group key is derived from target *values*, so a dataset refresh can move a key between regimes. Whatever is persisted should be understood as valid for a given data snapshot — worth recording the mirror revision alongside it.
- The grouped path is the scientifically stronger evaluation; the IID fallback is a consequence of the data (no replicates), not a design choice. The ticket makes that visible rather than changing it.
- Related: `_grouped_train_test_split` was non-deterministic across processes until RamanBench `24fdae5` (`str(frozenset)` hash-order in group labels). Any regime metadata is only meaningful for results produced at or after that commit.

**Comment by and18** [comment]
Related: I've recovered the actual replicate structure for four classification datasets and measured the effect — posted in #7 , in case it's useful input for the split_type decision here
